### ISSUE TITLE
Support for Trident API and other topologies

### ISSUE THREAD

**Issue by grinser** [colinsurprenant/redstorm#46]
Is there anything planned regarding the support of the newly introduced Trident API (Storm 0.8) as well as the other existing topologies?
I just updated the storm dependency of redstorm to storm 0.8.0 for my own testing and it seems that the existing redstorm samples work flawlessly.

**Comment by colinsurprenant** [comment]
Yes absolutely, I will make a release for 0.8.0 and plan to support Trident. AFAIK LinearDRPCTopology and  transactional topologies are being deprecated so I do not think it's worth spending time on it at this point, unless there is a request for it (I am not using them myself...). 

ACTUALLY, (... checking issues ...) there is a request for LinearDRPCTopology in issue #42 and I opened issue #41 for transactional topologies. doh. So, I might do #42. 

Thanks for testing the 0.8.0 upgrade! Good to know its working well!

**Comment by grinser** [comment]
Great that you plan to support 0.8.0 with redstorm.
Actually I wouldn't mind the old topologies as long as I could use all scenarios provided with the Trident API through Ruby code - this would be so awesome. Therefore I think the new ones should be of higher priority if you'd ask me ;-)

I know this is a tough one but do you have a vague timeline when you might work on it and release it?
I wish I could help with the porting but I'm still too new to all this stuff to totally grasp all the details.

**Comment by kerinin** [comment]
:+1: for Trident support!

**Comment by colinsurprenant** [comment]
I am leaving for vacations tonight :D  so I'll work on it at the end of August.

**Comment by mikeclymer** [comment]
I opened issue #42 before I was aware of the DRPC support with Trident. I don't think working on #42 is necessary at this point. Especially with the deprecation of LinearDRPCTopology. +1 for Trident support. Thank you for the great work.

**Comment by robmckinnon** [comment]
another :+1: for Trident support! :)

**Comment by colinsurprenant** [comment]
getting there, see issue #50

**Comment by schuyler** [comment]
bump

**Comment by benoist** [comment]
any news on this?

**Comment by colinsurprenant** [comment]
unfortunately not much but will reserve some time on this in the next few weeks.

**Comment by colinsurprenant** [comment]
I just pushed 0.7.0.beta1 which includes Trident examples in `examples/trident`. This is mainly using the Java API, a bit more work will be required to make this a bit cleaner, but it works!

Let me know what you think!

**Comment by benoist** [comment]
Looks good!
Is it also possible to create a custom aggregator instead of the built-in Sum and Count?

**Comment by colinsurprenant** [comment]
Yes it will shortly. Just need to create the Java proxies for all the `storm.trident.operation.*` interfaces ... for now only `Function` has been done.

**Comment by benoist** [comment]
Ok :)
I'll try this week to see how far it will get me with just the Function.

**Comment by ismith** [comment]
It'd be great to see example spec tests for a Trident topology, either here (testing examples/trident/word_count_topology) or in the redstorm-starter project.
